src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -208,106 +221,167 @@ contract SecurityCouncilManager is Initializable, ArbitrumTimelock, ISecurityCou
+    /// @param data Signed data
+    /// @param signature over data
+    /// @param account signer
+    /// @return true if signature is valid and from account
+    function _verify(bytes32 data, bytes memory signature, address account)
+        public
+        pure
+        returns (bool)
+    {
+        return data.toEthSignedMessageHash().recover(signature) == account;
     }
 
-    function getMarchCohort() external view returns (address[] memory) {
-        return marchCohort;
+    function getFirstCohort() external view returns (address[] memory) {
+        return firstCohort;
     }
 
-    function getSeptemberCohort() external view returns (address[] memory) {
-        return septemberCohort;
+    function getSecondCohort() external view returns (address[] memory) {
+        return secondCohort;
     }
 
-    /// @notice overridden; proposals can only be scheduled via scheduleUpdateMembers
-    function schedule(
-        address __,
-        uint256 ___,
-        bytes calldata ____,
-        bytes32 _____,
-        bytes32 ______,
-        uint256 _______
-    ) public override {
-        revert("SecurityCouncilManager: schedule not callable externally");
+    /// @notice generate unique salt for timelock scheduling
+    /// @param _members data to input / hash
+    function generateSalt(address[] memory _members) external view returns (bytes32) {
+        return keccak256(abi.encodePacked(_members, updateNonce));
     }
 
-    /// @notice overridden; proposals can only be scheduled via scheduleUpdateMembers
-    function scheduleBatch(
-        address[] calldata __,
-        uint256[] calldata ___,
-        bytes[] calldata ____,
-        bytes32 _____,
-        bytes32 ______,
-        uint256 _______
-    ) public override {
-        revert("SecurityCouncilManager: schedulebatch not callable externally");
+    /// @notice build new security council members array from first and second cohort and schedule a sequence of cross chain messages to update all security councils accordingly.
+    function _scheduleUpdate() internal {
+        // build array of security council members
+        address[] memory newMembers = new address[](secondCohort.length + firstCohort.length);
+        for (uint256 i = 0; i < firstCohort.length; i++) {
+            newMembers[i] = firstCohort[i];
+        }
+        for (uint256 i = 0; i < secondCohort.length; i++) {
+            newMembers[firstCohort.length + i] = secondCohort[i];
+        }
+
+        // build batch call to L1 timellck
+        uint256[] memory valuesForL1TimelockOpertaions = new uint256[](securityCouncils.length);
+        address[] memory targetsForL1TimelockOperations = new address[](securityCouncils.length);
+        bytes[] memory payloadsForL1TimelockOperations = new bytes[](securityCouncils.length);
+        for (uint256 i = 0; i < securityCouncils.length; i++) {
+            // values are all always 0
+            valuesForL1TimelockOpertaions[i] = 0;
```

Comment:
I think these are already set to 0, you could avoid creating a var here and do `bytes memory l1TimelockCallData = abi.encodeWithSelect(..., new uint256[](securityCouncils.length), ...)`